src/dwc_checklist.R

```diff
@@ -276,9 +277,75 @@ distribution %>%
 #' Save to CSV:
 write.csv(distribution, file = dwc_distribution_file, na = "", row.names = FALSE, fileEncoding = "UTF-8")
 
+#' ## Create speciesProfile extension
+#' 
+#' We use this extension to map the **salinity zone** information contained in `raw_salinity_zone` in the raw data file.
+#' `raw_salinity_zone` describes whether a species is found in brackish (B), freshwater (F), marine (M) or combined (B/M or F/B) salinity zone. 
+
+#' ### Pre-processing
+species_profile <- raw_data
+
+#' ### Term mapping
+#' 
+#' Map the source data to [Species Profile](http://rs.gbif.org/extension/gbif/1.0/speciesprofile.xml):
+
+#' #### taxonID
+species_profile %<>% mutate(taxonID = raw_taxonID)
+
+#' The following DwC terms from the Species Profile extension are used to map the `raw_salinity_zone` information: `isMarine` and `isFreshwater`.
+#' For completeness, we integrate `isTerrestrial` as this is an essential piece of information for the development of indicators for invasive species.
+#' This is how `raw_salinity_zone` maps to the three DwC terms:
+
+kable(as.data.frame(
+  matrix(data = c(
+    "F", "FALSE", "TRUE", "FALSE",
+    "M", "TRUE", "FALSE", "FALSE",
+    "B/M", "TRUE", "FALSE", "FALSE",
+    "F/B", "FALSE", "TRUE", "FALSE",
+    "B", "TRUE", "TRUE", "FALSE"),
+    nrow = 5, ncol = 4, byrow = T,
+    dimnames = list (c(1:5), c("salinity zone", "isMarine", "isFreshwater", "isTerrestrial"))
+  )))
+
+#' #### isMarine
+species_profile %<>% 
+  mutate(isMarine = case_when(
+    raw_salinity_zone == "M" | raw_salinity_zone == "B/M" | raw_salinity_zone == "B" ~ "TRUE",
```

Comment:
We could do that, but then there's more repetition in the code. I am in favour of using case_when, because this is actually when I think out loud when mapping the data (i.e. 'if the salinity zone of a species is marine, or B/M, or brackisch, than we say that isMarine = TRUE). But if you're not very familiar with R code, a recode() would maybe be better and easier to interpret. On the other hand, we use case_when() often enough... To conclude, I see arguments for both, so you can decide :-)

Reply:
Ok to leave as is.